src/App.jsx

```diff
@@ -1,13 +1,110 @@
-import { useState } from 'react'
-import './App.css'
+import React, { useState } from "react";
+import ClassCheckBox from "./components/ClassCheckbox";
+import classes from "./constants";
+import "./App.css";
 
-function App() {
+
+export default function App() {
+	const [data, setData] = useState(classes);
+
+	const handleClassCheck = (classIndex) => {
+		let dataClone = [...data];
+		let targetClass = dataClone[classIndex];
+
+		targetClass.checked = !targetClass.checked;
+		targetClass.sections = targetClass.sections.map((section) => {
+			let students = section.students.map((student) => ({
+				...student,
+				checked: targetClass.checked,
+			}));
+
+			return {
+				...section,
+				checked: targetClass.checked,
+				students,
+			};
+		});
+
+		dataClone[classIndex] = targetClass;
+		setData([...dataClone]);
+	};
+
+	const handleSectionCheck = (sectionIndex, classIndex) => {
+		let dataClone = [...data];
+		let sections = dataClone[classIndex].sections.map((section, index) => {
```

Comment:
Always add data availability check at these points. Since it will decrease the possibility of errors later on when the data is not available.

`let sections = dataClone[classIndex]?.sections?.map((section, index) => {`

Reply:
Sure will make the changes.